scipy/signal/_signaltools.py

```diff
@@ -2387,19 +2420,21 @@ def deconvolve(signal, divisor):
     array([ 0.,  1.,  0.,  0.,  1.,  1.,  0.,  0.])
 
     """
-    num = np.atleast_1d(signal)
-    den = np.atleast_1d(divisor)
+    xp = array_namespace(signal, divisor)
+
+    num = xpx.atleast_nd(signal, ndim=1, xp=xp)
+    den = xpx.atleast_nd(divisor, ndim=1, xp=xp)
```

Comment:
Nice that smoke-docs caught this! It demonstrates the need for additional array-like tests, though.
```suggestion
    num = xpx.atleast_nd(xp.asarray(signal), ndim=1, xp=xp)
    den = xpx.atleast_nd(xp.asarray(divisor), ndim=1, xp=xp)
```